### PR TITLE
ArrayCache integration

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Performance Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
ArrayCache integration

### CODE REVIEW

File: src/Storages/MarkCache.h [hunk 1]
```diff
@@ -19,25 +19,24 @@ namespace DB
...
+// struct MarksWeightFunction
+// {
+//     /// We spent additional bytes on key in hashmap, linked lists, shared pointers, etc ...
+//     static constexpr size_t MARK_CACHE_OVERHEAD = 128;
 
+//     size_t operator()(const MarksInCompressedFile & marks) const
+//     {
+//         return marks.size() * sizeof(MarkInCompressedFile) + MARK_CACHE_OVERHEAD;
+//     }
+// };
```
Comment (nikitamikhaylov) [inline src/Storages/MarkCache.h:31]: You maybe forgot to delete.

Comment (kitaisreal) [reply src/Storages/MarkCache.h]: Thanks. Fixed.

**Comment by alexey-milovidov** [comment]
Did not work out.

**Comment by al13n321** [comment]
What was the problem?

**Comment by alexey-milovidov** [comment]
Either slow or with bugs, but I don't remember.

**Pull request closed without merging**